evalml/tests/component_tests/test_components.py

```diff
@@ -1628,3 +1646,33 @@ def test_estimator_fit_respects_custom_indices(
     estimator.fit(X, y)
     pd.testing.assert_index_equal(X.index, X_original_index)
     pd.testing.assert_index_equal(y.index, y_original_index)
+
+
+def test_component_parameters_modifies_feature_target():
+    for component_class in all_components():
+        if (
+            issubclass(component_class, BaseSampler)
+            or issubclass(component_class, TargetTransformer)
+            or component_class in [TargetImputer]
+        ):
+            assert component_class.modifies_target
+        else:
+            assert not component_class.modifies_target
+        if issubclass(component_class, TargetTransformer) or component_class in [
+            TargetImputer
+        ]:
+            assert not component_class.modifies_features
+        else:
+            assert component_class.modifies_features
+
+
+def test_component_parameters_supported_by_list_API():
```

Comment:
Will smoke test in #2490. Right now, all components are still "supported" in list API, until it's deprecated in #2490 :)